packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/features/resolution/ResolutionFeature.java

```diff
@@ -235,20 +238,25 @@ private void configureResolution(ResolutionPreset resolutionPreset, int cameraId
       return;
     }
 
+    captureSize = null;
+
     if (Build.VERSION.SDK_INT >= 31) {
       recordingProfile =
-          getBestAvailableCamcorderProfileForResolutionPreset(cameraId, resolutionPreset);
+              getBestAvailableCamcorderProfileForResolutionPreset(cameraId, resolutionPreset);
       List<EncoderProfiles.VideoProfile> videoProfiles = recordingProfile.getVideoProfiles();
 
       EncoderProfiles.VideoProfile defaultVideoProfile = videoProfiles.get(0);
-      captureSize = new Size(defaultVideoProfile.getWidth(), defaultVideoProfile.getHeight());
-    } else {
+      if (defaultVideoProfile != null) {
+        captureSize = new Size(defaultVideoProfile.getWidth(), defaultVideoProfile.getHeight());
+      }
+    }
+
+    if (captureSize == null) {
```

Comment:
As a follow up to the comment I left below with regards to constructing `MediaRecorderBuilder` properly, you may want to ensure that either `recordingProfile` and `recordingProfileLegacy` is null. It may be easier to tell based on this whether `CamcorderProfile` or `EncoderProfiles` is being used as a result.